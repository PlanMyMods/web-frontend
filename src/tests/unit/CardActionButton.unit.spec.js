import { mount } from "@vue/test-utils";
import CardActionButton from "@/components/CardActionButton.vue";

describe("CardActionButton", () => {
  const text = "Button Text";
  const textColor = "text-gray-700";
  const textColorDark = "text-gray-300";
  const color = "bg-gray-200";
  const colorDark = "bg-gray-700";
  const hoverColor = "bg-gray-300";
  const hoverColorDark = "bg-gray-600";

  const activeBtnComponent = mount(CardActionButton, {
    propsData: {
      text: text,
      textColor: textColor,
      textColorDark: textColorDark,
      color: color,
      colorDark: colorDark,
      hoverColor: hoverColor,
      hoverColorDark: hoverColorDark,
      disabled: false,
    },
  });
  const buttonWrapper = activeBtnComponent.find("button");
  test("should match snapshot", () => {
    expect(activeBtnComponent.html()).toMatchSnapshot();
  });
  test("should have the correct text", () => {
    expect(activeBtnComponent.text()).toContain(text);
  });
  test("should have the correct button text color class", () => {
    expect(buttonWrapper.classes()).toContain(textColor);
  });
  test("should have the correct button text color class in dark mode", () => {
    expect(buttonWrapper.classes()).toContain(`dark:${textColorDark}`);
  });
  test("should have the correct button color class", () => {
    expect(buttonWrapper.classes()).toContain(color);
  });
  test("should have the correct button color class in dark mode", () => {
    expect(buttonWrapper.classes()).toContain(`dark:${colorDark}`);
  });
  test("should have the correct button hover color class", () => {
    expect(buttonWrapper.classes()).toContain(`hover:${hoverColor}`);
  });
  test("should have the correct button hover color class in dark mode", () => {
    expect(buttonWrapper.classes()).toContain(`dark:hover:${hoverColorDark}`);
  });
  test("should not have the cursor-not-allowed class when button not disabled", () => {
    expect(buttonWrapper.find("cursor-not-allowed").exists()).toBe(false);
  });
  test("should have the cursor-not-allowed class when disabled in dark mode", () => {
    const disabledBtn = mount(CardActionButton, {
      propsData: {
        text: text,
        textColor: textColor,
        color: color,
        hoverColor: hoverColor,
        disabled: true,
      },
    });
    expect(disabledBtn.find("button").classes()).toContain(
      "cursor-not-allowed"
    );
  });
});
