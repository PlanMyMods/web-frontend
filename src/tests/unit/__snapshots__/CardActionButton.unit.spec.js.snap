// Vitest Snapshot v1

exports[`CardActionButton > should match snapshot 1`] = `
"<div class=\\"mt-5\\"><button class=\\"bg-gray-200 dark:bg-gray-700 text-gray-700 dark:text-gray-300 font-bold rounded p-3 h-full w-full align-middle flex-auto hover:bg-gray-300 dark:hover:bg-gray-600\\">
    <div class=\\"flex relative w-full box-border align-baseline justify-center\\">Button Text</div>
  </button></div>"
`;
